### DISCUSSION TITLE
What are the differences between pgbackrest and pgmoneta?

### DISCUSSION THREAD

**Discussion by stephane-klein** [pgmoneta/pgmoneta#181] [Q&A]
What are the differences between [pgbackrest](https://pgbackrest.org/) and *pgmoneta*?

What are the differences in architecture?

When is it better to use pgmoneta rather than *pgbackrest*?

---
Edit:

I ask this question out of curiosity.
I am in a friendly state of mind.
I don't want to pit pgmoneta against pgbackrest or vice versa.
I have no problem that these two projects exist

**Comment by jesperpedersen** [comment]
There is no reason to put Open Source projects "against" each other. Each have their focus and development speed.

I think it is best for you to see how easy it is to see how to set them up, and - f.ex. - run them over a week. How is the disk space usage ? Are you getting the metrics you want (Prometheus) ? Is your data safe (AES) ? Is your crash recovery policy covered (CLI restore) ?

So, set it up, let it run, then start to crash your primary PostgreSQL instance, crash your backup server, crash the server where your archive and WAL shipping is located on, ... And, see how you recover.

Maybe the symlink support will work for you...

We value feedback, and our goals are high - and we will work on those.

All contributions - we mean all - are valued.

**Reply by stephane-klein** [reply]
> There is no reason to put Open Source projects "against" each other. Each have their focus and development speed.

I ask this question out of curiosity.  
I am in a friendly state of mind.  
I don't want to pit *pgmoneta* against *pgbackrest* or vice versa.  
I have no problem that these two projects exist

**Reply by jesperpedersen** [reply]
pgmoneta is part of a "family"

* https://agroal.github.io/pgagroal/
* https://pgmoneta.github.io/ (of course)
* https://pgexporter.github.io/

They share the same architecture; process based, and shared memory.

The overall goal is that these projects co-exists in the best possible way - but can work on their own. And, they are separate projects with different life cycles.

**Comment by jesperpedersen** [comment]
I see reasons why you should pick pgmoneta, and I see reasons - right now - that you should pick pgbackrest.

Your data is the king - there is no reason why you can't switch your backup solution. You need to make sure that your data is safe -- application layer, sure, but if your data is gone then where do you start from ?

**Reply by Its-Alex** [reply]
@jesperpedersen Hi there! 👋 I totally get what you're saying. After diving into the readme files of both projects, I'm having a bit of trouble figuring out the key differences and goals. Could this be what @stephane-klein was curious about?

It seems crucial to understand how data backups are handled to be aware of the drawbacks of each tool and determine which situations suit my environment best. For these reasons, the questions above seem significant to me.

**Reply by jesperpedersen** [reply]
Well, right now we are investigating how https://github.com/pgmoneta/pgmoneta/issues/147 is best handled. @Jubilee101 is working on that - pgbackrest have a solution. Is that important in certain cases, yes - in all, no.

Each solution have different features at certain point - f.ex. if you want Prometheus integration then pgmoneta got that integrated in the main project already...

We will welcome a comparison, but just know that people have different feature sets that they are looking for. And, that is why we welcome contributions !

I think in the end (many years from now) all backup solutions will have a similar feature set.
